sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1114,7 +1114,7 @@ class SessionCatalog(
       }
       externalCatalog.dropFunction(db, name.funcName)
     } else if (!ignoreIfNotExists) {
-      throw new NoSuchFunctionException(db = db, func = identifier.toString)
+      throw new NoSuchPermanentFunctionException(db = db, func = identifier.toString)
```

Comment:
Actually, Catalog-wise, seems the exception makes sense. It's not in temp and permanent registries. Why don't you just catch the exception and fix it in the drop command?

Reply:
![image](https://user-images.githubusercontent.com/51401130/63145056-846a9580-c013-11e9-9f05-8964c83a1cd5.png)
In catalog dropTempFunction is present to handle temp function .
in dropTempFunction  - **NoSuchTempFunctionException** exception is getting thrown
both functions throws different exceptions , is the required to change in  dropTempFunction?

Reply:
No, I meant the exception describes what the codes do in a way. Looking back the history, seems it was updated at https://github.com/apache/spark/commit/5c6b0855787c080d3e233eb09c05c025395e7cb3. OK, let's fix it.